test/logging/timestamp_checkpointing_test.cpp

```diff
@@ -0,0 +1,161 @@
+      "upid2), "
+      // not supported yet "UNIQUE (value4, value5), "
+      "PRIMARY KEY (pid1, pid2));";
+  TestingSQLUtil::ExecuteSQLQuery(constraint_test_sql);
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO checkpoint_constraint_test VALUES (1, 2, 3, 4, 0, 1, 2);");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO checkpoint_constraint_test VALUES (5, 6, 7, 8, 1, 6, 7);");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO checkpoint_constraint_test VALUES (9, 10, 11, 12, 2, 11, "
+      "12);");
+  TestingSQLUtil::ExecuteSQLQuery("COMMIT;");
+
+  // insert test
+  TestingSQLUtil::ExecuteSQLQuery("BEGIN;");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO checkpoint_table_test VALUES (3, 0.0, 'xxxx');");
+  TestingSQLUtil::ExecuteSQLQuery("COMMIT;");
+
+  // output created table information to verify checkpoint recovery
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  auto catalog = catalog::Catalog::GetInstance();
+  auto storage = storage::StorageManager::GetInstance();
+  auto default_db_catalog = catalog->GetDatabaseObject(DEFAULT_DB_NAME, txn);
+  for (auto table_catalog_pair : default_db_catalog->GetTableObjects()) {
+    auto table_catalog = table_catalog_pair.second;
+    auto table = storage->GetTableWithOid(table_catalog->GetDatabaseOid(),
+                                          table_catalog->GetTableOid());
+    LOG_INFO("Table %d %s\n%s", table_catalog->GetTableOid(),
+    		table_catalog->GetTableName().c_str(), table->GetInfo().c_str());
+
+    for (auto column_pair : table_catalog->GetColumnObjects()) {
+      auto column_catalog = column_pair.second;
+      auto column =
+          table->GetSchema()->GetColumn(column_catalog->GetColumnId());
+      LOG_INFO("Column %d %s\n%s", column_catalog->GetColumnId(),
+      		column_catalog->GetColumnName().c_str(), column.GetInfo().c_str());
+    }
+  }
+  txn_manager.CommitTransaction(txn);
+
+  // generate table and data that will be out of checkpointing.
+  TestingSQLUtil::ExecuteSQLQuery("BEGIN;");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO checkpoint_table_test VALUES (4, -1.0, 'out of the "
+      "checkpoint');");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO checkpoint_table_test VALUES (5, -2.0, 'out of the "
+      "checkpoint');");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "CREATE TABLE out_of_checkpoint_test (pid INTEGER PRIMARY KEY);");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO out_of_checkpoint_test VALUES (1);");
+  // TestingSQLUtil::ExecuteSQLQuery("CREATE DATABASE out_of_checkpoint;");
+
+  // do checkpointing
+  checkpoint_manager.StartCheckpointing();
+
+  EXPECT_TRUE(checkpoint_manager.GetStatus());
+
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+  checkpoint_manager.StopCheckpointing();
+
+  TestingSQLUtil::ExecuteSQLQuery("COMMIT;");
+
+  EXPECT_FALSE(checkpoint_manager.GetStatus());
+
```

Comment:
It would be better if we can read the file back and see if it is what we expect.

Reply:
I added the test case to make sure the created table information and the inserted values are correct.